ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -78,27 +159,29 @@ fi
 # Get desired compute node count
 getDesiredComputeCount
 
-# wait for all nodes to reach Ready=true to ensure that all machines and nodes came up, before we run
-# any e2e tests that might require specific workload capacity.
-echo "$(date) - waiting for all nodes to be ready in $READY_WAIT_TIMEOUT"
+# Check if this is a HCP cluster
+is_hcp_cluster="$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r ".hypershift.enabled")"
+log "hypershift.enabled is set to $is_hcp_cluster"
+
 ret=0
-oc wait nodes --all --for=condition=Ready=true --timeout="$READY_WAIT_TIMEOUT" || ret=$?
+echo "Wait for all nodes to be ready and schedulable."
+waitForReady "$desired_compute_count" || ret=$?
 
 if [[ "$ret" == 0 ]]; then
     # Get count of worker only nodes in Ready state
-    node_count="$(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/SchedulingDisabled!=,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type=='Ready')].status.conditions[-1].type}" | wc -w | xargs)"
+    node_count="$(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type=='Ready')].status.conditions[-1].type}" | wc -w | xargs)"
     echo "Count of worker only nodes in Ready state: $node_count"
-
+    
     # Check worker node count matches requested replica count
     if (( "$node_count" >= "$desired_compute_count" )); then
         echo "$(date): All $node_count worker nodes are ready and match desired $desired_compute_count node count."
-    else
-        echo "$(date): $node_count worker nodes are ready but does not match desired $desired_compute_count node count in $READY_WAIT_TIMEOUT."
         listNodeDetails
-        exit 1
+    else
+        echo "$(date): $node_count worker nodes are ready but does not match desired $desired_compute_count node count."
+        listDetails
     fi
 else
-    echo "Some or all nodes failed to become ready in $READY_WAIT_TIMEOUT."
-    listNodeDetails
+    echo "Failed to wait for nodes to be ready. Return code: $ret."
+    echo "Exiting test"
```

Comment:
should we still list out some node information here for the failure

Reply:
This is when we fail to run the function itself or if the function fails for any reason.

Reply:
I didn't add any details here purposefully yet. Was hoping to see a failure when we hit this and then decide best how to handle it and add additional details as needed.